plugins/crd/crd_queue.go

```diff
@@ -113,10 +113,10 @@ func workforever(plugin *Plugin, queue workqueue.RateLimitingInterface, informer
 					plugin.Log.Errorf("Informer cached version of object (%s/%s) different than worker queue version", namespace, name)
 				}
 				plugin.Log.Infof("Got most up to date version of '%s/%s'. Syncing...", namespace, name)
-				plugin.objectStore.ObjectCreated(message.(objectMessage).obj)
+				plugin.ObjectStore.ObjectCreated(message.(objectMessage).obj)
```

Comment:
Why ObjectStore now is public? The whole idea of Interface was to protect data from other plugins accessing directly and give them only methods they need to access data???

Reply:
@sbezverk Its a Dependency ... if you make it private, nobody can wire in an alternate implementation that meets the API.

Reply:
One other thing to consider when thinking about this... if you are accessing this from a variable of the type of the interface... Deps is *not* accessible, and thus ObjectStore is not accessible.  It's only if you access it from a variable of type of the CRD struct that you can get a hold of Deps and thus ObjectStore.

Reply:
I think it should be excluded from `regular` dependencies. Each plugin would identify if it needs it and then Call SharedPlugin to get interfaces to talk to it. Data consistency here is the key and it can only be provided by hiding the actual data and providing secured interface to access it.

Reply:
@sbezverk SharedPlugin will *always* return the implementation from the objectstore package (as it should).

Reply:
So... the Dep is stored using the ObjectStore interface, so if someone wants to much with it, the only things they can do are:

a)  Replace it with an alternate ObjectStore (which they can always do with UseDeps() anyway) or 
b) Use the ObjectStore interface

If they want to do #b, then they could just call objectstore,SharedPlugin() and get the ObjectStore that way and muck with it.

I don't quite see how this increases exposure...

Reply:
My preference would be for the interface to be public and for the implementation to be swappable. Deps would allow for us to mock specific state and greatly increase the testability of the CRD implementation.

Reply:
FWIW, I agree with @fkautz here. I like the interface being public with the implementation being swappable. For now, objectstore is an in-memory map of objects, but it's conceivable to think we may want it to be something else in the medium term.